libgraphics/src/legacygradients.cpp

```diff
@@ -339,7 +340,7 @@ class MCGGeneralizedGradientShader : public SkShader
             {
                 if ((int)m_buffer_width < count * GRADIENT_AA_SCALE)
                 {
-                    uindex_t t_size = m_buffer_width * GRADIENT_AA_SCALE * GRADIENT_AA_SCALE;
+                    uindex_t t_size;
```

Comment:
This needs to be initialized with the current size of the array, as MCMemoryResizeArray clears anything between old size and new size (so the existing value of t_size is used).

Reply:
Maybe MCMemoryReallocate would be more appropriate then, as the buffer doesn't need to be initialized before use

Reply:
@runrevmark I've reinstated the size initialization, however the original calculation was incorrect so i've replaced it with a corrected version